pandas/io/formats/format.py

```diff
@@ -565,315 +569,414 @@ def __init__(
-        Render a DataFrame to a list of columns (as lists of strings).
+    def _truncate_horizontally(self) -> None:
+        """Remove columns, which are not to be displayed and adjust formatters.
+
+        Attributes affected:
+            - tr_frame
+            - formatters
+            - tr_col_num
         """
-        # this method is not used by to_html where self.col_space
-        # could be a string so safe to cast
-        col_space = {k: cast(int, v) for k, v in self.col_space.items()}
+        assert self.max_cols_fitted is not None
+        col_num = self.max_cols_fitted // 2
+        if col_num >= 1:
+            cols_to_keep = [
+                x
+                for x in range(self.frame.shape[1])
+                if x < col_num or x >= len(self.frame.columns) - col_num
+            ]
+            self.tr_frame = self.tr_frame.iloc[:, cols_to_keep]
 
-        frame = self.tr_frame
-        # may include levels names also
+            # truncate formatter
+            if isinstance(self.formatters, (list, tuple)):
+                slicer = itemgetter(*cols_to_keep)
+                self.formatters = slicer(self.formatters)
+        else:
+            col_num = cast(int, self.max_cols)
+            self.tr_frame = self.tr_frame.iloc[:, :col_num]
+        self.tr_col_num = col_num
 
-        str_index = self._get_formatted_index(frame)
+    def _truncate_vertically(self) -> None:
+        """Remove rows, which are not to be displayed.
+
+        Attributes affected:
+            - tr_frame
+            - tr_row_num
+        """
+        assert self.max_rows_fitted is not None
+        row_num = self.max_rows_fitted // 2
+        if row_num >= 1:
+            rows_to_keep = [
+                x
+                for x in range(self.frame.shape[0])
+                if x < row_num or x >= len(self.frame) - row_num
+            ]
+            self.tr_frame = self.tr_frame.iloc[rows_to_keep, :]
+        else:
+            row_num = cast(int, self.max_rows)
+            self.tr_frame = self.tr_frame.iloc[:row_num, :]
+        self.tr_row_num = row_num
+
+    def _get_strcols_without_index(self) -> List[List[str]]:
+        strcols: List[List[str]] = []
 
         if not is_list_like(self.header) and not self.header:
-            stringified = []
-            for i, c in enumerate(frame):
+            for i, c in enumerate(self.tr_frame):
                 fmt_values = self._format_col(i)
                 fmt_values = _make_fixed_width(
-                    fmt_values, self.justify, minimum=col_space.get(c, 0), adj=self.adj
+                    strings=fmt_values,
+                    justify=self.justify,
+                    minimum=int(self.col_space.get(c, 0)),
+                    adj=self.adj,
                 )
-                stringified.append(fmt_values)
+                strcols.append(fmt_values)
+            return strcols
+
+        if is_list_like(self.header):
+            assert isinstance(self.header, list)
```

Comment:
is_list_like is not the same as isinstance(..., list). can you revert

Reply:
Good catch. Thank you